src/mat/4C_mat_fourier.cpp

```diff
@@ -0,0 +1,203 @@
+Mat::Fourier::Fourier(Mat::PAR::Fourier* params) : params_(params) {}
+
+/*----------------------------------------------------------------------*
+ *----------------------------------------------------------------------*/
+void Mat::Fourier::pack(Core::Communication::PackBuffer& data) const
+{
+  // pack type of this instance of ParObject
+  int type = unique_par_object_id();
+  add_to_pack(data, type);
+
+  int matid = params_->id();
+  add_to_pack(data, matid);
+}
+
+/*----------------------------------------------------------------------*
+ *----------------------------------------------------------------------*/
+void Mat::Fourier::unpack(Core::Communication::UnpackBuffer& buffer)
+{
+  Core::Communication::extract_and_assert_id(buffer, unique_par_object_id());
+
+  // matid
+  int matid;
+  extract_from_pack(buffer, matid);
+  params_ = nullptr;
+  if (Global::Problem::instance()->materials() != nullptr)
+    if (Global::Problem::instance()->materials()->num() != 0)
+    {
+      const int probinst = Global::Problem::instance()->materials()->get_read_from_problem();
+      Core::Mat::PAR::Parameter* mat =
+          Global::Problem::instance(probinst)->materials()->parameter_by_id(matid);
+
+      FOUR_C_ASSERT_ALWAYS(mat->type() == material_type(),
+          "Type of parameter material %d does not fit to calling type %d", mat->type(),
+          material_type());
+
+      params_ = static_cast<Mat::PAR::Fourier*>(mat);
+    }
+}
+
+/*----------------------------------------------------------------------*
+ *----------------------------------------------------------------------*/
+void Mat::Fourier::evaluate(const Core::LinAlg::Matrix<1, 1>& gradtemp,
+    Core::LinAlg::Matrix<1, 1>& cmat, Core::LinAlg::Matrix<1, 1>& heatflux) const
+{
+  // conductivity tensor in 1d is always a scalar quantity
+  cmat(0, 0) = params_->conduct_[0];
+
+  // heatflux
+  heatflux.multiply_nn(cmat, gradtemp);
+}
+
+/*----------------------------------------------------------------------*
+ *----------------------------------------------------------------------*/
+void Mat::Fourier::evaluate(const Core::LinAlg::Matrix<2, 1>& gradtemp,
+    Core::LinAlg::Matrix<2, 2>& cmat, Core::LinAlg::Matrix<2, 1>& heatflux) const
+{
+  // conductivity tensor in 2d is a 2x2 matrix with either constant or variable values on the
+  // diagonal or a full matrix.
+  cmat.clear();
+  switch (params_->conduct_para_num_)
+  {
+    case 1:  // scalar value is given
+    {
+      cmat(0, 0) = params_->conduct_[0];
+      cmat(1, 1) = params_->conduct_[0];
+      break;
+    }
+    case 2:  // diagonal values are given
+    {
+      cmat(0, 0) = params_->conduct_[0];
+      cmat(1, 1) = params_->conduct_[1];
+      break;
+    }
+    case 4:  // full tensor is given
+    {
+      cmat(0, 0) = params_->conduct_[0];
+      cmat(0, 1) = params_->conduct_[1];
+      cmat(1, 0) = params_->conduct_[2];
+      cmat(1, 1) = params_->conduct_[3];
+      break;
+    }
```

Comment:
obviously, the `CONDUCT_PARA_NUM` parameter is to cover the non-diagonal thermal conductivity tensor, rather than the dimension. In that case, I would name it `CONDUCT_TYPE` or similar.

Reply:
I took the name `THERM_PARA_NUM` as example from another material.